com.unity.perception/Runtime/GroundTruth/Labelers/Visualization/OverlayPanel.cs

```diff
@@ -0,0 +1,231 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using UnityEngine.UI;
+
+namespace UnityEngine.Perception.GroundTruth
+{
+    /// <summary>
+    /// Interface that should be defined by a class that wants to be able to provide a image to the overlay
+    /// panel.
+    /// </summary>
+    public interface IOverlayPanelProvider
+    {
+        /// <summary>
+        /// Provides an image to the overlay panel.
+        /// </summary>
+        /// <returns>The image for the overlay panel.</returns>
+        Texture GetOverlayImage();
+    }
+
+    /// <summary>
+    /// Some labeler's result in a full screen image per frame. The overlay panel controls which of these labeler's image
+    /// is currently shown.
+    /// </summary>
+    public class OverlayPanel : MonoBehaviour
+    {
+        bool m_Enabled;
+        Dictionary<string, IOverlayPanelProvider> m_OverlayProviders = new Dictionary<string, IOverlayPanelProvider>();
+
+        GUIStyle m_LabelStyle;
+        GUIStyle m_SliderStyle;
+        GUIStyle m_SelectorToggleStyle;
+        GUIStyle m_WindowStyle;
+
+        float m_SegmentTransparency = 0.8f;
+        float m_BackgroundTransparency;
+
+        GameObject m_SegCanvas;
+        GameObject m_SegVisual;
+        RawImage m_OverlayImage;
+
+        int m_CachedHeight;
+        int m_CachedWidth;
+
+        bool m_ShowPopup = false;
+        Texture2D m_NormalDropDownTexture;
+        Texture2D m_HoverDropDownTexture;
+
+        /// <summary>
+        /// Registers a labeler as an overlay provider.
+        /// </summary>
+        /// <param name="provider">The overlay provider to register.</param>
+        /// <typeparam name="T">The type of the labeler, which must be an overlay provider and camera labeler</typeparam>
+        /// <returns>Did the labeler register properly</returns>
+        public bool RegisterOverlayProvider<T>(T provider) where T : CameraLabeler, IOverlayPanelProvider
```

Comment:
So right now, we can't remove labelers at runtime. The UI is tied to the camera so adding/removing cameras should work. But I did rework code to (I think) support this case in the future.

Anyway, I got rid of registering overlay panels and I just check the labelers that perception has if they are overlay panel providers.

Reply:
Actually it is completely legitimate to remove labelers via script. The only reason it's not allowed in the UI is due to a bug with serialization. I just think we should confirm that this will not cause issues and then move on.

Reply:
Ok, tested it to make sure it works with labelers coming on- and off-line. A couple changes were made to support it.